ignite/metrics/clustering/calinski_harabasz_score.py

```diff
@@ -0,0 +1,107 @@
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(features, labels)``
+            or ``{'features': features, 'labels': labels}``.
+        check_compute_fn: if True, ``compute_fn`` is run on the first batch of data to ensure there are no
+            issues. If issues exist, user is warned that there might be an issue with the ``compute_fn``.
+            Default, True.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+        skip_unrolling: specifies whether output should be unrolled before being fed to update method. Should be
+            true for multi-output model, for example, if ``y_pred`` contains multi-ouput as ``(y_pred_a, y_pred_b)``
+            Alternatively, ``output_transform`` can be used to handle this.
+
+    Examples:
+        To use with ``Engine`` and ``process_function``, simply attach the metric instance to the engine.
+        The output of the engine's ``process_function`` needs to be in format of
+        ``(features, labels)`` or ``{'features': features, 'labels': labels, ...}``.
+
+        .. include:: defaults.rst
+            :start-after: :orphan:
+
+        .. testcode::
+
+            metric = CalinskiHarabaszScore()
+            metric.attach(default_evaluator, "calinski_harabasz_score")
+            X = torch.tensor([
+                    [-1.04, -0.71, -1.42, -0.28, -0.43],
+                    [0.47, 0.96, -0.43, 1.57, -2.24],
+                    [-0.62, -0.29, 0.10, -0.72, -1.69],
+                    [0.96, -0.77, 0.60, -0.89, 0.49],
+                    [-1.33, -1.53, 0.25, -1.60, -2.0],
+                    [-0.63, -0.55, -1.03, -0.89, -0.77],
+                    [-0.26, -1.67, -0.24, -1.33, -0.40],
+                    [-0.20, -1.34, -0.52, -1.55, -1.50],
+                    [2.68, 1.13, 2.51, 0.80, 0.92],
+                    [0.33, 2.88, 1.35, -0.56, 1.71]
+            ])
+            Y = torch.tensor([0, 0, 0, 0, 1, 1, 1, 1, 2, 2])
+            state = default_evaluator.run([{"features": X, "labels": Y}])
+            print(state.metrics["calinski_harabasz_score"])
+
+        .. testoutput::
+
+            5.733935121807529
+
+    .. versionadded:: 0.5.2
+    """
+
+    def __init__(
+        self,
+        output_transform: Callable[..., Any] = lambda x: x,
+        check_compute_fn: bool = True,
+        device: Union[str, torch.device] = torch.device("cpu"),
+        skip_unrolling: bool = False,
+    ) -> None:
+        try:
+            from sklearn.metrics import calinski_harabasz_score  # noqa: F401
+        except ImportError:
+            raise ModuleNotFoundError("This module requires scikit-learn to be installed.")
```

Comment:
Checking for the required package in init is a good idea!
Especially, that we may need to do the same for other epoch-like metrics: https://github.com/pytorch/ignite/blob/master/ignite/metrics/precision_recall_curve.py etc

Reply:
By the way, I saw that in Kohen Kappa implementation you were doing: 
https://github.com/pytorch/ignite/blob/302c707f79dbef50a3920baf11b76eba5ee4200e/ignite/metrics/cohen_kappa.py#L74C9-L93

Can we pick a single pattern for example like in https://github.com/pytorch/ignite/blob/master/ignite/metrics/average_precision.py and apply it everywhere?
If we agree with that can you please send a separate PR for Kohen Kappa fix please